test/integration_test.py

```diff
@@ -2060,17 +2068,31 @@ def test_response_security_headers(self):
                     else:
                         response = requests.get(str(endpoint / path))
                         response.raise_for_status()
-                        expected = AzulChaliceApp.security_headers | expected_headers
-                        # FIXME: Add a CSP header with a nonce value to text/html responses
-                        #        https://github.com/DataBiosphere/azul-private/issues/6
+                        expected = AzulChaliceApp.security_headers() | expected_headers
+                        nonces = set()
+                        for directive in response.headers['Content-Security-Policy'].split(';'):
+                            match = directive_re.fullmatch(directive)
+                            self.assertIsNotNone(match)
+                            name, value = match.groups()
+                            if name in ('script-src', 'style-src'):
+                                match = nonce_re.search(value)
+                                if match is not None:
+                                    nonce = match.group(1)
+                                    # Our nonce token is 32 bytes Base64 encoded
+                                    self.assertEqual(43, len(nonce))
```

Comment:
As I stated before, you can roll this into the regex. I don't understand why you went back to this approach.

Reply:
This was necessary to confirm that the endpoints not expected to return a nonce value, do not. Otherwise it would only be confirming that the endpoints don't return a nonce value of 43 characters length.

Reply:
> Otherwise the regex would only be confirming that the endpoints don't return a nonce value of 43 characters length.

Well, the same argument applies to your current approach. As currently written, entries starting in `nonce-` but followed a non-base64 character would slip through.

Reply:
> As currently written, entries starting in nonce- but followed a non-base64 character would slip through.

It is true they would not be matched by the RE `r"'nonce-([a-zA-Z0-9_-]+)'"`, however the full CSP header value is still being included in the expected values and would fail the test if the response header contained an unexpected nonce.

Reply:
> and the tests would fail if

Please prove that with a script-src that has a nonce with a 43 character base64 nonce followed by a nonce that contains a character that is non base64 but is allowed in CSP values in general.

The basic smell I am objecting to is that you are asserting the base64-ness of the nonce in the regex but the length in a separate assert. Both are additional requirements imposed by *us*, not the spec. This is confusing and, even if were able to prove your assertion, obscure and brittle. In fact, the code that parses the CSP should adhere to the RFC. The RFC does not require base64 or a certain length.